python/pyspark/sql/connect/dataframe.py

```diff
@@ -1344,9 +1344,9 @@ def collect(self) -> List[Row]:
         if self._session is None:
             raise Exception("Cannot collect on empty session.")
         query = self._plan.to_proto(self._session.client)
-        table = self._session.client.to_table(query)
+        table, schema = self._session.client.to_table(query)
 
-        schema = from_arrow_schema(table.schema)
+        schema = schema or from_arrow_schema(table.schema)
```

Comment:
when I was implementing the collection I was thinking about this dumb question: is it possible to make arrow schema 100% compatible with spark schema if we store the different fields in arrow metadata?
e.g. for a spark `MapType`, store the `valueContainsNull` as a metadata in the `pa.schema`.

Reply:
That's interesting and I was thinking about the similar thing, but I didn't take the approach because it takes some space in the `RecordBatch` that could repeatedly be sent to client. That means the schema space could be huge if repeat many times.

Reply:
Btw, for the collections, we can retrieve nullability from the schema.

For example:

```
if pa.is_list(at):
  field = at.value_field
  ArrayType(from_arrow_type(field.type), containsNull=field.nullable)
```

Reply:
It seems that MapType's `valueContainsNull` is stored in `to_arrow_type` but discarded in `from_arrow_type`.